src/react-apps/ux-editor/src/sagas/api/apiSagas.ts

```diff
@@ -237,6 +246,10 @@ function* apiCheckValue(connectionDef: any, lastUpdatedDataBinding: any, lastUpd
                 if (!updatedComponent) {
                   // This space intentionally left blank
                 } else {
+                  if (isPartOfRepeatingGroup) {
+                    updatedDataBinding = { ...updatedDataBinding };
+                    updatedDataBinding.DataBindingName = updatedDataBinding.DataBindingName.replace(dataModelGroup, dataModelGroupWithIndex);
```

Comment:
Are you sure that this doesn't actually update the data model in the store?

Reply:
Since the updatedDataBinding.DataBindingName is not an object we are altering a copy and not a reference, which is intended. Have confirmed by manually checking the state.